webview.h

```diff
@@ -496,6 +499,7 @@ class gtk_webkit_engine {
   }
   virtual ~gtk_webkit_engine() = default;
   void *window() { return (void *)m_window; }
+  int step(int blocking) { return gtk_main_iteration_do(blocking); }
```

Comment:
The original `webview_loop` function returned a private variable `should_exit`, which was managed by webview rather than GTK. Do you think that'd be appropriate?

https://github.com/webview/webview/blob/0bee6e66e4ac33af2ed44cb5c5ff9f7e9ddd0b56/webview.h#L82

Reply:
I did some exploration and came up with this, which seems to more closely mimic old webview behavior. I don't think we need to change `run` like this, but calling `terminate` needs to be able to stop the loop.
```
-  int step(int blocking) { return gtk_main_iteration_do(blocking); }
-  void run() { gtk_main(); }
-  void terminate() { gtk_main_quit(); }
+  int step(int blocking) { 
+    gtk_main_iteration_do(blocking);
+    return should_exit;
+  }
+  void run() {
+    while (step(0) == 0) {}
+  }
+  void terminate() { should_exit = 1; } }
```

Reply:
`terminate` should still be able to exit the loop without the `should_exit` property as it already does so internally. [`gtk_main_iteration_do`](https://docs.gtk.org/gtk3/func.main_iteration_do.html) is essentially the same as [`gtk_main`](https://docs.gtk.org/gtk3/func.main.html) except it only does one iteration of the main loop, returning whether [`gtk_main_quit`](https://docs.gtk.org/gtk3/func.main_quit.html) has been called.

Reply:
So the answer to your question is no as it is redundant, the reason (I think) for how it was before was that MacOS requires the `should_exit` variable and it just got copied over.